README.md

```diff
@@ -104,15 +113,98 @@ None.
 
 ## Q&A
 
-**Q**: Why not a deep-copy option?
+### Why not a deep-copy option?
+
+`JSON.stringify` aside, it is not common for builtin JavaScript APIs to traverse arbitrary objects deeply.
+
+`Array.prototype.flat` is _deep_, but only for the well defined boundaries of arrays.
+
+### Why only own keys?
+
+This follows other builtins such as `Object.keys` and also matches https://github.com/tc39/proposal-joint-iteration.
+
+### What about symbol keys?
+
+All enumerable properties are used, included enumerable symbols.
+
+This matches https://github.com/tc39/proposal-joint-iteration.
+
+This does differ from [existing solutions](#existing-solutions), which follow `Object.keys` semantics (ignoring symbols). This difference is not perceived to be an issue due to the low usage of own enumerable symbols.
+
+## Alternatives considered
+
+### `Promise.ownProperties`
+
+```javascript
+const {
+  shape,
+  color,
+  mass,
+} = await Promise.ownProperties({
+  shape: getShape(),
+  color: getColor(),
+  mass: getMass(),
+});
+```
+
+### `Promise.fromEntries`
 
-**A**: Deep copy traditionally has been left out of JavaScript for a number of reasons.  While a recursive promise walking API might be possible, combining it with this feels overcomplicated and unlikely to pass TC39's smell tests.
+```javascript
+const {
+  shape,
+  color,
+  mass,
+} = await Promise.fromEntries(Object.entries({
+  shape: getShape(),
+  color: getColor(),
+  mass: getMass(),
+}));
+```
 
-**Q**: Why not all keys?
+### `Promise.all` overload
 
-**A**: This implies walking the prototype chain.  Note this proposal would have been `Promise.allProperties()` originally, but that naming confuses what the polyfill does above with that intended use case.  Plus, if a prototype has promises on it, then how would we safely construct the prototype of the resulting object?  Going down the prototype chain opens a can of worms.
+Dispatch depending if the argument is an iterable or not.
```

Comment:
If you're going to mention this, probably worth saying why it was rejected: namely, discomfort with the shape of the output depending on the shape of the input and the risk of accidentally passing multiple arguments instead of an array.